src/DynamoCore/Graph/Workspaces/HomeWorkspaceModel.cs

```diff
@@ -347,6 +395,78 @@ internal virtual void OnSetNodeDeltaState(DeltaComputeStateEventArgs e)
             historicalTraceData = copiedData;
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HomeWorkspaceModel"/> class
+        /// by given information about it and specified item collections
+        /// </summary>
+        /// <param name="engine"><see cref="EngineController"/> object assosiated with this home workspace
+        /// to coordinate the interactions between some DesignScript sub components.</param>
+        /// <param name="scheduler"><see cref="DynamoScheduler"/> object to add tasks in queue to execute</param>
+        /// <param name="factory">Node factory to create nodes</param>
+        /// <param name="traceData">Preloaded trace data</param>
+        /// <param name="nodes">Node collection of the workspace</param>
+        /// <param name="notes">Note collection of the workspace</param>
+        /// <param name="annotations">Group collection of the workspace</param>
+        /// <param name="presets">Preset collection of the workspace</param>
+        /// <param name="elementResolver">ElementResolver responsible for resolving 
+        /// a partial class name to its fully resolved name</param>
+        /// <param name="info">Information for creating custom node workspace</param>
+        /// <param name="verboseLogging">Indicates if detailed descriptions should be logged</param>
+        /// <param name="isTestMode">Indicates if current code is running in tests</param>
+        /// <param name="linterManager">The linter manager from the DynamoModel that owns this workspace</param>
+        public HomeWorkspaceModel(EngineController engine,
+            DynamoScheduler scheduler,
+            NodeFactory factory,
+            IEnumerable<KeyValuePair<Guid, List<CallSite.RawTraceData>>> traceData,
+            IEnumerable<NodeModel> nodes,
+            IEnumerable<NoteModel> notes,
+            IEnumerable<AnnotationModel> annotations,
+            IEnumerable<PresetModel> presets,
+            ElementResolver resolver,
+            WorkspaceInfo info,
+            bool verboseLogging,
+            bool isTestMode,
+            LinterManager linterManager)
+            : base(nodes, notes, annotations, info, factory, presets, resolver, linterManager)
+        {
+            Debug.WriteLine("Creating a home workspace...");
```

Comment:
please don't copy and paste this, if if it has to exist in two spots refactor it out into a shared private method, this section of code is confusing enough.